controllers/organization_controller_sync.go

```diff
@@ -0,0 +1,123 @@
+package controllers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	orgv1 "github.com/appuio/control-api/apis/organization/v1"
+	controlv1 "github.com/appuio/control-api/apis/v1"
+	"github.com/vshn/appuio-keycloak-adapter/keycloak"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var orgImportAnnot = "keycloak-adapter.vshn.net/importing"
+
+// Sync lists all Keycloak groups in the realm and creates corresponding Organizations if they do not exist
+func (r *OrganizationReconciler) Sync(ctx context.Context) error {
+	log := log.FromContext(ctx)
+
+	gs, err := r.Keycloak.ListGroups(ctx)
+	if err != nil {
+		log.Error(err, "error listing Keycloak groups")
+		return err
+	}
+
+	orgMap, err := r.getOrganiztionMap(ctx)
+	if err != nil {
+		log.Error(err, "error listing Organizations")
+		return err
+	}
+
+	var groupErr error
+	recordError := func(descr string, g keycloak.Group, err error) {
+		if groupErr == nil {
+			groupErr = errors.New("")
+		}
+		log.WithValues("group", g).Error(err, descr)
+		groupErr = fmt.Errorf("%w\n%s: %s\n    %s", groupErr, g.Name, descr, err.Error())
+	}
+	for _, g := range gs {
+		org, ok := orgMap[g.Name]
+		if !ok {
+			log.V(1).WithValues("group", g).Info("creating organization")
+			org, err = r.startImportOrganizationFromGroup(ctx, g)
+			if err != nil {
+				recordError("failed to start organization import", g, err)
+				continue
+			}
+		}
+		if org.Annotations[orgImportAnnot] == "true" {
+			log.V(1).WithValues("group", g).Info("updating organization members")
+			err := r.updateOrganizationMembersFromGroup(ctx, g)
+			if err != nil {
+				recordError("failed to update organization members", g, err)
+				continue
+			}
+			err = r.finishImportOrganizationFromGroup(ctx, org)
+			if err != nil {
+				recordError("failed to finish organization import", g, err)
+				continue
+			}
+		}
+	}
+	if groupErr != nil {
+		return fmt.Errorf("partial sync failure:\n%w", groupErr)
+	}
+	return nil
+}
+
+func (r *OrganizationReconciler) getOrganiztionMap(ctx context.Context) (map[string]*orgv1.Organization, error) {
```

Comment:
nitpick: I would rename `get` to `fetch` so that it's more clear that some resources are being loaded
```suggestion
func (r *OrganizationReconciler) fetchOrganiztionMap(ctx context.Context) (map[string]*orgv1.Organization, error) {
```